tensorflow_serving/tools/docker/Dockerfile.devel-mkl

```diff
@@ -0,0 +1,110 @@
+FROM ubuntu:16.04 as base_build
+
+ARG TF_SERVING_VERSION_GIT_BRANCH=master
+ARG TF_SERVING_VERSION_GIT_COMMIT=head
+
+LABEL maintainer="Karthik Vadla <karthik.vadla@intel.com>"
+LABEL tensorflow_serving_github_branchtag=${TF_SERVING_VERSION_GIT_BRANCH}
+LABEL tensorflow_serving_github_commit=${TF_SERVING_VERSION_GIT_COMMIT}
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        automake \
+        build-essential \
+        ca-certificates \
+        curl \
+        git \
+        libcurl3-dev \
+        libfreetype6-dev \
+        libpng12-dev \
+        libtool \
+        libzmq3-dev \
+        mlocate \
+        openjdk-8-jdk\
+        openjdk-8-jre-headless \
+        pkg-config \
+        python-dev \
+        software-properties-common \
+        swig \
+        unzip \
+        wget \
+        zip \
+        zlib1g-dev \
+        && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN curl -fSsL -O https://bootstrap.pypa.io/get-pip.py && \
+    python get-pip.py && \
+    rm get-pip.py
+
+RUN pip --no-cache-dir install \
+    grpcio \
+    h5py \
+    keras_applications \
+    keras_preprocessing \
+    mock \
+    numpy \
+    requests
+
+# Set up Bazel
+# Need >= 0.15.0 so bazel compiles work with docker bind mounts.
+ENV BAZEL_VERSION 0.15.0
+WORKDIR /
+RUN mkdir /bazel && \
+    cd /bazel && \
+    curl -H "User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36" -fSsL -O https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel-$BAZEL_VERSION-installer-linux-x86_64.sh && \
+    curl -H "User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36" -fSsL -o /bazel/LICENSE.txt https://raw.githubusercontent.com/bazelbuild/bazel/master/LICENSE && \
+    chmod +x bazel-*.sh && \
+    ./bazel-$BAZEL_VERSION-installer-linux-x86_64.sh && \
+    cd / && \
+    rm -f /bazel/bazel-$BAZEL_VERSION-installer-linux-x86_64.sh
+
+# Download MKL sources
+RUN git clone https://github.com/01org/mkl-dnn.git && \
+    cd mkl-dnn && \
+    cd scripts && ./prepare_mkl.sh && cd ..
+
+# Download TF Serving sources (optionally at specific commit).
+WORKDIR /tensorflow-serving
+RUN git clone --branch=${TF_SERVING_VERSION_GIT_BRANCH} https://github.com/tensorflow/serving . && \
+    git remote add upstream https://github.com/tensorflow/serving.git && \
+    if [ "${TF_SERVING_VERSION_GIT_COMMIT}" != "head" ]; then git checkout ${TF_SERVING_VERSION_GIT_COMMIT} ; fi
+
+
+FROM base_build as binary_build
+
+# Build, and install TensorFlow Serving
+# Building with -march=sandybridge will allow the container to run on as many platforms as possible;
+# Intel(R) MKL-DNN will still take advantage of all modern CPU instructions at runtime.
+ARG TF_SERVING_BUILD_OPTIONS="--define=using_mkl=true --copt=-march=native --copt=-O3 --host_copt=-march=native -c opt"
```

Comment:
These args should be just --config=mkl and specified in the bazel.rc file, so it can be built easily outside of the dockerfile as well.

Reply:
or else i will document to pass this is `--build-arg TF_SERVING_BUILD_OPTIONS="--define=using_mkl=true --copt=-march=native --copt=-O3 --host_copt=-march=native -c opt"` during `docker build`. 

If you want to stick to `.bazelrc` route let me know. I don't see any any `.bazelrc` related stuff in your actual `Dockerfile`. I'm very new `bazel` related stuff, if you guide me through it happy to work. 
@gautamvasudevan

Reply:
Yes, see how it looks in the Dockerfile.devel piece. You should be able to use `--config=mkl --config=nativeopt` instead.

Reply:
Sure will work on it. All i have to do is use `ARG TF_SERVING_BUILD_OPTIONS="--config=mkl"` which does bazel build with `MKL` optimizations. 

I don't see anywhere you building `tensorflow` in the dockerfile. Where  is that happening?
 and which version of `tensorflow` is used by `tensorflow-seving`? @gautamvasudevan

Reply:
Tensorflow is built as a [dependency in bazel](https://github.com/tensorflow/serving/blob/master/tensorflow_serving/model_servers/BUILD). The version used for releases maps 1:1 usually (so 1.12 TF Serving uses 1.12 of TF). However, for the master build, we build off of the latest green build of TF (which is usually at or very near master).